components/list/list-item-drag-handle.js

```diff
@@ -0,0 +1,196 @@
+				padding: 0;
+				width: 0.9rem;
+			}
+			/* Firefox includes a hidden border which messes up button dimensions */
+			button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-list-item-drag-handle-dragger-button {
+				background-color: unset;
+			}
+			button,
+			button[disabled]:hover,
+			button[disabled]:focus {
+				background-color: var(--d2l-color-gypsum);
+				color: var(--d2l-color-ferrite);
+			}
+			button:hover,
+			button:focus {
+				background-color: var(--d2l-color-mica);
+			}
+			button[disabled] {
+				cursor: default;
+				opacity: 0.5;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.keyboardActive = false;
+	}
+
+	focus() {
+		const node = getFirstFocusableDescendant(this);
+		if (node) {
+			node.focus();
+		}
+	}
+
+	render() {
+		return html`${this.keyboardActive ? this.renderKeyboardDragging() : this.renderDragger()}`;
+	}
+
+	renderDragger() {
+		return html`
+			<button class="d2l-list-item-drag-handle-dragger-button" @click="${this._handleDeactiveKeyboard}" @keydown="${this._handleDeactiveKeyboard}" aria-label="${this.text}">
+				<d2l-icon icon="tier1:dragger"></d2l-icon>
+			</button>
+		`;
+	}
+
+	renderKeyboardDragging() {
+		return html`
+			<button class="d2l-list-item-drag-handle-keyboard-button" @blur="${this._onBlur}" @keydown="${this._handleActiveKeyboard}" aria-label="${this.text}">
+				<d2l-icon class="sub-icon" icon="tier1:arrow-toggle-up" @click="${this._dispatchActionUp}"></d2l-icon>
+				<d2l-icon class="sub-icon" icon="tier1:arrow-toggle-down" @click="${this._dispatchActionDown}"></d2l-icon>
+			</button>
+		`;
+	}
+
+	_dispatchAction(action) {
+		if (!action) return;
+		this.dispatchEvent(new CustomEvent('d2l-list-item-drag-handle-action', {
+			detail: { action },
+			bubbles: false
+		}));
+	}
+
+	_dispatchActionDown() {
+		this._dispatchAction(dragActions.down);
+	}
+
+	_dispatchActionUp() {
+		this._dispatchAction(dragActions.up);
+	}
+
+	_onBlur() {
+		this.keyboardActive = false;
+		if (!this._tabbing) {
+			this._dispatchAction(dragActions.save);
+		}
+	}
+
+	_handleActiveKeyboard(e) {
+		if (!this.keyboardActive) {
+			return;
+		}
+		let action = null;
+		switch (e.keyCode) {
+			case keyCodes.UP:
+				action = dragActions.up;
+				break;
+			case keyCodes.DOWN:
+				action = dragActions.down;
+				break;
+			case keyCodes.HOME:
+				action = dragActions.first;
+				break;
+			case keyCodes.END:
+				action = dragActions.last;
+				break;
+			case keyCodes.TAB:
+				action = e.shiftKey ? dragActions.previousElement : dragActions.nextElement;
+				break;
+			case keyCodes.ESC:
+				action = dragActions.cancel;
+				this.updateComplete.then(() => this.keyboardActive = false);
+				break;
+			case keyCodes.ENTER:
+			case keyCodes.SPACE:
+			case keyCodes.RIGHT:
+				action = dragActions.save;
+				this.updateComplete.then(() => this.keyboardActive = false);
+				break;
+			default:
+				return;
+		}
+		this._dispatchAction(action);
+		e.preventDefault();
```

Comment:
What is the behavior for a long list for a keyboard user?  Are they able to "drag" the item to a list item that is below the fold?

Reply:
I need to test this.

Reply:
So I tested and it does scroll with the movement of the card. Although, I was testing in the prototype currently up on my site lord site. We should make a note to test this when we have a list that actually moves.